### ISSUE TITLE
Page to blog: issue links failed to load

### ISSUE THREAD

**Issue by moondog13** [chris-kent/Confluence-Blog-Tools#6]
I was ecstatic to see this plugin as a way to create blog templates for confluence as well as convert a lot of our prior made pages to blog entries. I tested this out on some of our pages which utilize the jira macro to link the page to jira issues. After converting a confluence page to a blog a new issue link is created for the jira issue. The issue link displays as "Page Failed to load" however when clicking on the new link the confluence page is properly displayed. [Here is a screenshot.](https://www.intersafeit.com/pubcloud/index.php/s/pLWegTwDEmJO5HJ) Is this a known shortfall and is there something I can do to fix this? 

Also is there a way to bulk convert pages in a space to blog entries given none of the pages have children?

**Comment by chris-kent** [comment]
Hi there,

I tried your scenario, I had a slightly different behaviour in JIRA. My
JIRA created a new "mentions" link correctly with the blog name, but still
had the old link, which I could safely delete in JIRA.

You say that your "mentions" link is recycled to the new blog page, but has
the wrong text on the link... I can't explain this and must be a JIRA
thing.  What version of JIRA do you have?

In any case, I am not sure how to fix this.  Maybe try a re-index in JIRA
administration???  Worst case is to delete the link and add another one.
Sounds painful if you have a few to do. Speaking of which, doing bulk moves
is not supported - sorry. How many pages do you need to convert? Maybe a
client side script could be created if you know all the pageid's... I could
investigate this for you.

Under the covers, Moving between Pages and Blogs is unfortunately a bit
frustrating as blogs and pages are not compatible entities, so I have to
literally make new content and move individual components from the page to
the blog.  Hence once the move is finished, the old Page is deleted and
hence any links to it are invalid.

Cheers
Chris

**Comment by chris-kent** [comment]
Hi

I had a thought for a better work around. Try this approach

1. Delete the mentions link in jira
2. Move the page using the plugin tool

This may give you a cleaner result.

Cheers

**Comment by chris-kent** [comment]
I was also thinking of an easier way to do bulk conversions.

If all the pages where children pages of a master page. I could easily add
an option to convert all children pages too.

So you would need to reorder the pages together. Then run the tool over the
parent and children.

Would this work for you?
On 5 May 2016 10:07 pm, "Chris Kent" <mr.chris.kent@gmail.com> wrote:

Hi there,

I tried your scenario, I had a slightly different behaviour in JIRA. My
JIRA created a new "mentions" link correctly with the blog name, but still
had the old link, which I could safely delete in JIRA.

You say that your "mentions" link is recycled to the new blog page, but has
the wrong text on the link... I can't explain this and must be a JIRA
thing.  What version of JIRA do you have?

In any case, I am not sure how to fix this.  Maybe try a re-index in JIRA
administration???  Worst case is to delete the link and add another one.
Sounds painful if you have a few to do. Speaking of which, doing bulk moves
is not supported - sorry. How many pages do you need to convert? Maybe a
client side script could be created if you know all the pageid's... I could
investigate this for you.

Under the covers, Moving between Pages and Blogs is unfortunately a bit
frustrating as blogs and pages are not compatible entities, so I have to
literally make new content and move individual components from the page to
the blog.  Hence once the move is finished, the old Page is deleted and
hence any links to it are invalid.

Cheers
Chris

**Comment by chris-kent** [comment]
Hi,

The convert all Children pages was implemented around Version 3.2, but please upgrade to the latest version and this functionality will be available.  Simply group all the pages to convert under when single Parent and Convert the Parent page, checking "All Children" and wait.

Hope this works for you.

Cheers
